### ISSUE TITLE
Dynamic Links C++ SDK crash on iOS when opening the app with a Dynamic Link

### ISSUE THREAD

**Issue by ForestRingGames** [firebase/quickstart-cpp#21]
Hello Firebase-Team,

We've just faced a crash in our app after clicking on a dynamic link. The crash seems to happen inside the Firebase library. Attached you find a crash-screenshot with a stacktrace.

<img width="2672" alt="bildschirmfoto 2018-01-28 um 19 11 12" src="https://user-images.githubusercontent.com/21338444/35485392-6a394560-045f-11e8-8fe7-b0dfa74621dc.png">

**Comment by jonsimantov** [comment]
Thanks for posting this! We'll investigate and report back ASAP.

**Comment by stewartmiles** [comment]
Hi @ForestRingGames , we tracked this down to the error reporting path of the Dynamic Links  iOS library.  In the case where you use a custom URI scheme and the URI scheme (e.g my.cool.app://) mismatches the bundle ID of your application (e.g my.cool.apps://), the iOS Dynamic Links library will call the completion block in of FIRDynamicLinks:handleUniversalLink:completion: with a nil NSError and a nil FIRDynamicLink i.e in the following code `dynamic_link` and `error` end up both being nil as it can't parse the link...

```
[[FIRDynamicLinks dynamicLinks] handleUniversalLink:url
                                         completion:^(FIRDynamicLink *_Nullable dynamic_link, NSError *_Nullable error) {
    // completion logic
  }];
```

the code in the C++ iOS library incorrectly assumed that if `error` is nil then `dynamic_link` would be non-nil, bad assumption.  We have a fix lined up for our next release.  In the meantime if you're using custom schemes you'll need to wait for our release as we'll end up trying to decode them via dynamic links and hit this error.  Otherwise, you should check your application for custom scheme handlers, [this page](https://developer.apple.com/documentation/uikit/core_app/communicating_with_other_apps_using_custom_urls) describes where this located in Xcode.

**Comment by ForestRingGames** [comment]
@stewartmiles Thank you that's awesome that you found out something! :)

So that means - after the next release this will not happen again, right? Regardless of what URI types, we've been using now?
Do you already have an estimate when the build will be out and/or do you have some kind of email notification that I can sign up for to definitely know I need to update now? :)

**Comment by stewartmiles** [comment]
@ForestRingGames the build is likely to land next week.  Alas, we don't have a way to subscribe for new release updates 👎.  But what we - the team - do is go through all resolved issues on public forums like this (firebase-talk@ etc.) and post when we have a release that resolves an issue.  So hang out on this thread and we'll send a follow up when a new release is posted on https://firebase.google.com/support/release-notes/cpp-relnotes

**Comment by johnb003** [comment]
This is fixed in Firebase C++ 4.4.2. https://firebase.google.com/support/release-notes/cpp-relnotes#4.4.2
